### PR TITLE
postgres fulltext support

### PR DESCRIPTION
**Pull request by yrong**
* sample usage  about where condition 

```
{keyword:{$fulltext:"whatever"}}
```

[postgres full text reference](https://www.postgresql.org/docs/9.6/static/textsearch-controls.html)

_Thanks for wanting to fix something on Sequelize - we already love you long time! Please delete this text and fill in the template below. If unsure about something, just do as best as you're able._

_If your PR only contains changes to documentation, you may skip the template below._

### Pull Request check-list

_Please make sure to review and check all of these items:_

- [x] Does `npm run test` or `npm run test-DIALECT` pass with this change (including linting)?
- [ ] Does your issue contain a link to existing issue (Closes #[issue]) or a description of the issue you are solving?
- [ ] Have you added new tests to prevent regressions?
- [ ] Is a documentation update included (if this change modifies existing APIs, or introduces new ones)?
- [ ] Have you added an entry under `Future` in the changelog?

_NOTE: these things are not required to open a PR and can be done afterwards / while the PR is open._

### Description of change

_Please provide a description of the change here._

### CODE REVIEW

**Comment by yrong** [comment]
```
plainto_tsquery([ config regconfig, ] querytext text) returns tsquery
```
I use [zhparse](https://github.com/amutu/zhparser) as config here and I think it does need some extra configuration in postgres.

```
SELECT plainto_tsquery('zh_parse', '你好，朋友');

"'你好' & '朋友'"
```

Yes, I admit it is not a perfect solution but something better than nothing.

**Comment by yrong** [comment]
@janmeier 

further modification per your suggestion, let me know if any problem.

File: package.json [hunk 1]
```diff
@@ -1,5 +1,5 @@
 {
-  "name": "sequelize",
+  "name": "sequelize-fulltext-search",
```
Comment (janmeier) [inline package.json]: Please don't change the name of the package :)

File: lib/dialects/abstract/query-generator.js [hunk 2]
```diff
@@ -2110,6 +2112,9 @@ const QueryGenerator = {
             } else if (value && value.$col) {
               escapeValue = false;
               value = this.whereItemQuery(null, value);
+            } else if(value && key==='$fulltext'){
```
Comment (janmeier) [inline lib/dialects/abstract/query-generator.js]: Spacing around the = signs

File: lib/dialects/abstract/query-generator.js [hunk 3]
```diff
@@ -2110,6 +2112,9 @@ const QueryGenerator = {
...
+                escapeValue = false;
+                value = this.fullTextSearchQuery(value,this.options);
```
Comment (janmeier) [inline lib/dialects/abstract/query-generator.js]: Space after comma

File: lib/dialects/postgres/query-generator.js [hunk 4]
```diff
@@ -889,6 +889,10 @@ const QueryGenerator = {
     }
 
     return AbstractQueryGenerator.setAutocommitQuery.call(this, value, options);
+  },
+
+  fullTextSearchQuery(value,options) {
+    return `plainto_tsquery('${options.regconfig||'english'}','${value}')`
```
Comment (janmeier) [inline lib/dialects/postgres/query-generator.js]: value needs to be escaped using `this.escape`

File: test/unit/dialects/postgres/query-generator.test.js [hunk 5]
```diff
@@ -527,6 +527,12 @@ if (dialect.match(/^postgres/)) {
           arguments: ['myTable', {where: {field: {not: 3}}}],
           expectation: 'SELECT * FROM myTable WHERE myTable.field != 3;',
           context: {options: {quoteIdentifiers: false}}
+        },{
+          title: 'generate fulltext search',
+          arguments: ['myTable', {where: {field: {$fulltext: 'foo'}}}],
+          expectation: 'SELECT * FROM myTable WHERE myTable.field @@ plainto_tsquery(\'english\',\'foo\');',
```
Comment (janmeier) [inline test/unit/dialects/postgres/query-generator.test.js]: Use a `"`  here , to avoid having to escape the `'`

Comment (yrong) [inline test/unit/dialects/postgres/query-generator.test.js]: change per review

**Comment by yrong** [comment]
change per review

File: package.json [hunk 6]
```diff
@@ -1,7 +1,7 @@
...
   "description": "Multi dialect ORM for Node.JS",
-  "version": "0.0.0-development",
+  "version": "4.1.0",
```
Comment (eseliger) [inline package.json]: revert this please, we use it for semantic-release

File: lib/dialects/postgres/query-generator.js [hunk 7]
```diff
@@ -904,6 +904,10 @@ const QueryGenerator = {
     }
 
     return AbstractQueryGenerator.setAutocommitQuery.call(this, value, options);
+  },
+
+  fullTextSearchQuery(value,options) {
+    return `plainto_tsquery(${this.escape(options.regconfig||'english')},${this.escape(value)})`
```
Comment (eseliger) [inline lib/dialects/postgres/query-generator.js]: semicolon

File: lib/dialects/postgres/query-generator.js [hunk 8]
```diff
@@ -904,6 +904,10 @@ const QueryGenerator = {
     }
 
     return AbstractQueryGenerator.setAutocommitQuery.call(this, value, options);
+  },
+
+  fullTextSearchQuery(value,options) {
```
Comment (eseliger) [inline lib/dialects/postgres/query-generator.js]: `(value, options) {`

File: test/unit/dialects/postgres/query-generator.test.js [hunk 9]
```diff
@@ -502,6 +502,12 @@ if (dialect.match(/^postgres/)) {
...
+          context: {options: {quoteIdentifiers: false,regconfig:'english'}},
```
Comment (eseliger) [inline test/unit/dialects/postgres/query-generator.test.js]: space

**Pull request closed without merging**